great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,392 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional
+
+from great_expectations.core._docs_decorators import public_api
+from great_expectations.core.batch_spec import BatchSpec, PathBatchSpec
+from great_expectations.core.util import AzureUrl, DBFSPath, GCSUrl, S3Url
+from great_expectations.experimental.datasources.data_asset.data_connector.batch_filter import (
+    BatchFilter,
+    build_batch_filter,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.data_connector import (
+    DataConnector,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.regex_parser import (
+    RegExParser,
+)
+
+# TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+# TODO: <Alex>ALEX</Alex>
+# from great_expectations.experimental.data_asset.data_connector.sorter import Sorter
+# TODO: <Alex>ALEX</Alex>
+from great_expectations.experimental.datasources.data_asset.data_connector.util import (
+    batch_definition_matches_batch_request,
+    map_batch_definition_to_data_reference_string_using_regex,
+    map_data_reference_string_to_batch_definition_list_using_regex,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+class DataConnectorStorageDataReferenceResolver:
+    DATA_CONNECTOR_NAME_TO_STORAGE_NAME_MAP: Dict[str, str] = {
+        "InferredAssetS3DataConnector": "S3",
+        "ConfiguredAssetS3DataConnector": "S3",
+        "InferredAssetGCSDataConnector": "GCS",
+        "ConfiguredAssetGCSDataConnector": "GCS",
+        "InferredAssetAzureDataConnector": "ABS",
+        "ConfiguredAssetAzureDataConnector": "ABS",
+        "InferredAssetDBFSDataConnector": "DBFS",
+        "ConfiguredAssetDBFSDataConnector": "DBFS",
+    }
```

Comment:
Should this be an enum?

Reply:
@Kilo59 It is a map -- or are you saying that some of the keys/values?  I am sure that there is lots of opportunity for improving efficiency/readability, it is just from the perspective of this PR, I would prefer for this to stay as is, since this is a direct copy from V3 and works (at least for now).  We can iterate on these improvements as we gear it toward launch.  Thanks.

Reply:
@Kilo59 This entire class was deleted -- the ZEP design makes it superfluous.  Here is the reasoning.

As @donaldheppner pointed this to me, the `Datasource` may have all the information needed to guide the `DataConnector` (without the `ExecutionEngine` or its name being involved in any way).  This turned out to be correct.

The entire `DataConnectorStorageDataReferenceResolver` which is critical in V3 is superfluous in ZEP, because the very lookup criteria in V3 — “the Storage-Engine tuple” are exactly the ZEP characteristics for `Datasource` types (which is what I learned from you and @NathanFarmer this morning, and the code from Nathan is already merged).  

While in V3, we had to provide this mapping:
```
        (
            "S3",
            "PandasExecutionEngine",
        ),
        (
            "S3",
            "SparkDFExecutionEngine",
        ),
        (
            "GCS",
            "PandasExecutionEngine",
        ),
        (
            "GCS",
            "SparkDFExecutionEngine",
        ),
        (
            "ABS",
            "PandasExecutionEngine",
        ),
        (
            "ABS",
            "SparkDFExecutionEngine",
        ),
        (
            "DBFS",
            "SparkDFExecutionEngine",
        ),
        (
            "DBFS",
            "PandasExecutionEngine",
        ),
```

in ZEP, these tuple components are exactly the syllables comprising the ZEP Datasource names:
```
* S3PandasDatasource
* S3SparkDatasource
* GCSPandasDatasource
* GCSSparkDatasource
* ABSPandasDatasource
* ABSSparkDatasource
```

and so on.  Hence, no mapping is needed; the `Datasource` supplies the mapping function (please see `great_expectations/core/util.py` for the templates that map file paths to fully-qualified object references).